### PR TITLE
refactor: couple of improvements

### PR DESCRIPTION
**Pull request by anonrig**
Before:

```
GHz=6.17 cycle/byte=15.2723 cycles/url=1.12182k instructions/byte=69.255 instructions/cycle=4.53468 instructions/ns=27.979 instructions/url=5.08709k ns/url=181.818 speed=295.053M/s time/byte=3.38922ns time/url=248.953ns url/s=4.01681M/s
```

After:

```
GHz=6.298 cycle/byte=15.5891 cycles/url=1.14509k instructions/byte=69.078 instructions/cycle=4.43117 instructions/ns=27.9075 instructions/url=5.07409k ns/url=181.818 speed=299.983M/s time/byte=3.33352ns time/url=244.862ns url/s=4.08393M/s
```

### CODE REVIEW

File: src/parser.cpp [hunk 1]
```diff
@@ -277,7 +278,6 @@ namespace ada::parser {
 
             // If c is U+003F (?), then set url’s query to the empty string, and state to query state.
             if ((input_position != input_size) && (url_data[input_position] == '?')) {
-              url.query = "";
```
Comment (lemire) [inline src/parser.cpp:280]: This is fine, but in general, url.query.clear() is effectively free. It is true that url.query = "" is expensive.

**Review by lemire — APPROVED** [review]
Good!

File: src/parser.cpp (hunk 1 above)
Comment (lemire) [reply src/parser.cpp:280]: `clear()` is just like a couple of instructions... very, very fast.

**Comment by lemire** [comment]
Merging.

File: src/parser.cpp (hunk 1 above)
Comment (anonrig) [reply src/parser.cpp:280]: This was unnecessary because we are already assigning a new value in the query state

**Pull request merged**